### PR TITLE
Remove vBAT promotions support

### PR DESCRIPTION
**Pull request by zenparsing**
Resolves https://github.com/brave/brave-browser/issues/36582

## Submitter Checklist:

- [ ] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

**Review by mkarolin — APPROVED** [review]
`strings`++

**Review by Douglashdaniel — APPROVED** [review]
++ Wallet Front-end

File: ios/browser/api/brave_rewards/brave_rewards_api.mm [hunk 1]
```diff
@@ -1197,17 +1040,14 @@ - (void)startNotificationTimers {
...
-                                       selector:@selector
-                                       (checkForNotificationsAndFetchGrants)
+                                       selector:@selector(checkForNotifications)
                                        userInfo:nil
                                         repeats:NO];
   });
 }
 
-- (void)checkForNotificationsAndFetchGrants {
+- (void)checkForNotifications {
```
Comment (kylehickinson) [inline ios/browser/api/brave_rewards/brave_rewards_api.mm:1049]: This entire method can just be deleted (along with the NSTimer created that periodically calls it) now that its not going to do anything in here anymore. The `lastNotificationCheckDate` property can also be removed

**Review by SergeyZhukovsky — APPROVED** [review]
Android ++

File: ios/browser/api/brave_rewards/brave_rewards_api.mm (hunk 1 above)
Comment (zenparsing) [reply ios/browser/api/brave_rewards/brave_rewards_api.mm]: Done.

File: chromium_presubmit_config.json5 [hunk 2]
```diff
@@ -148,6 +147,7 @@
     // A temporary fix to silence a failing presubmit check on file renames.
     "CheckStableMojomChanges": [
       "components/services/bat_ads/public/interfaces/bat_ads\\.mojom",
+      "components/brave_rewards/common/mojom/rewards_engine\\.mojom",
```
Comment (emerick) [inline chromium_presubmit_config.json5:150]: Can we keep these alphabetized?

**Review by emerick — APPROVED** [review]
LGTM

File: android/java/org/chromium/chrome/browser/rewards/BraveRewardsPanel.java [hunk 3]
```diff
@@ -650,15 +631,13 @@ private void showNotification(String id, int type, long timestamp, String[] args
...
-        mClaimInProcess = mBraveRewardsNativeWorker.isGrantClaimInProcess();
-        if (mClaimInProcess) {
-            BraveRewardsHelper.crossfade(actionNotificationButton, claimProgress, View.GONE, 1f,
-                    BraveRewardsHelper.CROSS_FADE_DURATION);
-        } else {
-            actionNotificationButton.setEnabled(true);
-            BraveRewardsHelper.crossfade(claimProgress, actionNotificationButton, View.GONE, 1f,
-                    BraveRewardsHelper.CROSS_FADE_DURATION);
-        }
+        actionNotificationButton.setEnabled(true);
```
Comment (deeppandya) [inline android/java/org/chromium/chrome/browser/rewards/BraveRewardsPanel.java:634]: we can remove the change here. as it won't be needed. i will remove it in the follow up issue.

File: browser/brave_rewards/android/brave_rewards_native_worker.cc [hunk 4]
```diff
@@ -562,7 +549,7 @@ void BraveRewardsNativeWorker::OnGetCurrentBalanceReport(
   JNIEnv* env = base::android::AttachCurrentThread();
   if (report) {
     std::vector<double> values;
-    values.push_back(report->grants);
+    values.push_back(0);
```
Comment (deeppandya) [inline browser/brave_rewards/android/brave_rewards_native_worker.cc:552]: optional : i think we can just remove above.

**Review by deeppandya — APPROVED** [review]
LGTM. Thank you @zenparsing for the clean up

**Review by kylehickinson — APPROVED** [review]
iOS++ Probably more to be removed in the future

File: android/java/org/chromium/chrome/browser/rewards/BraveRewardsPanel.java (hunk 3 above)
Comment (zenparsing) [reply android/java/org/chromium/chrome/browser/rewards/BraveRewardsPanel.java:634]: I'll leave this for for a follow-up.

File: browser/brave_rewards/android/brave_rewards_native_worker.cc (hunk 4 above)
Comment (zenparsing) [reply browser/brave_rewards/android/brave_rewards_native_worker.cc]: Done. I also had to rename the index constants in `BraveRewardsPanel.java`.

File: chromium_presubmit_config.json5 (hunk 2 above)
Comment (zenparsing) [reply chromium_presubmit_config.json5:149]: done

File: components/brave_rewards/core/state/state_keys.h [hunk 5]
```diff
@@ -11,8 +11,6 @@ namespace state {
 
 const char kServerPublisherListStamp[] = "publisher_prefix_list_stamp";
 const char kUpholdAnonAddress[] = "uphold_anon_address";  // DEPRECATED
-const char kPromotionLastFetchStamp[] = "promotion_last_fetch_stamp";
```
Comment (szilardszaloki) [inline components/brave_rewards/core/state/state_keys.h:14]: Should we rather just deprecate these?

File: components/brave_rewards/core/contribution/contribution.cc [hunk 6]
```diff
@@ -551,14 +551,9 @@ void Contribution::OnEntrySaved(const std::string& contribution_id,
 
   const std::string& queue_id = queue->id;
 
-  if (wallet_type == constant::kWalletUnBlinded) {
```
Comment (szilardszaloki) [inline components/brave_rewards/core/contribution/contribution.cc:554]: I think we can now get rid of this wallet type, too (`components/brave_rewards/core/global_constants.h`)?

File: components/brave_rewards/core/state/state_keys.h (hunk 5 above)
Comment (zenparsing) [reply components/brave_rewards/core/state/state_keys.h:14]: They are deprecated on the browser side. I don't think we need to keep around this string, do you?

File: components/brave_rewards/core/contribution/contribution.cc [hunk 7]
```diff
@@ -852,9 +842,8 @@ void Contribution::Retry(mojom::ContributionInfoPtr contribution,
 
   switch (contribution->processor) {
     case mojom::ContributionProcessor::BRAVE_TOKENS: {
-      RetryUnblindedContribution({mojom::CredsBatchType::PROMOTION},
-                                 std::move(result_callback),
-                                 std::move(contribution));
+      // TODO(zenparsing): Remove this enum value?
```
Comment (szilardszaloki) [inline components/brave_rewards/core/contribution/contribution.cc:845]: `ContributionProcessor::BRAVE_TOKENS`? I think so.

File: components/brave_rewards/core/wallet/wallet_balance.cc [hunk 8]
```diff
@@ -23,25 +23,8 @@ WalletBalance::WalletBalance(RewardsEngineImpl& engine) : engine_(engine) {}
...
-  if (!engine_->state()->GetVBatExpired()) {
-    for (const auto& token : tokens) {
-      total += token->value;
-    }
-  }
-
   auto balance = mojom::Balance::New();
-  balance->total = total;
+  balance->total = 0;
   balance->wallets.emplace(constant::kWalletUnBlinded, balance->total);
```
Comment (szilardszaloki) [inline components/brave_rewards/core/wallet/wallet_balance.cc:28]: Is this still needed somewhere?

File: components/brave_rewards/core/state/state_keys.h (hunk 5 above)
Comment (szilardszaloki) [reply components/brave_rewards/core/state/state_keys.h:14]: Right — no, that's fine. 👍🏻

**Review by szilardszaloki — APPROVED** [review]
`components/brave_rewards/core` LGTM

File: components/brave_rewards/core/contribution/contribution.cc (hunk 7 above)
Comment (zenparsing) [reply components/brave_rewards/core/contribution/contribution.cc]: Thanks for pointing this out. I created a follow-up issue to remove these and other remaining vBAT references:

https://github.com/brave/brave-browser/issues/36936

File: components/brave_rewards/core/wallet/wallet_balance.cc (hunk 8 above)
Comment (zenparsing) [reply components/brave_rewards/core/wallet/wallet_balance.cc:28]: Will be removed in https://github.com/brave/brave-browser/issues/36936

File: components/brave_rewards/core/contribution/contribution.cc (hunk 6 above)
Comment (zenparsing) [reply components/brave_rewards/core/contribution/contribution.cc:554]: Will remove in https://github.com/brave/brave-browser/issues/36936

**Pull request merged**